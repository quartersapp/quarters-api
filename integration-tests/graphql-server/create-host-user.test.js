/* eslint-env mocha */

const request = require('supertest')
const app = require('lib/app').listen()
const { User } = require('lib/db/models')
const { truncateModel, enableSnapshots } = require('test-helpers')

describe('integration-tests/graphql-server/create-host-user', () => {
  beforeEach(() => truncateModel(User))
  beforeEach(enableSnapshots)

  it('can create a new host user', async () => {
    const variables = {
      input: {
        firstName: 'Lucas',
        lastName: 'McLaughlin',
        email: 'lucas.m@gmail.com',
        password: 'pass1234',
        hostBio: 'test host bio',
        hostGooglePlaceId: 'test-google-place-id'
      }
    }

    const query = `
      mutation CreateHostUser($input: CreateHostUserInput!) {
        createHostUser(input: $input) {
          user {
            name,
            email
          }
        }
      }
    `

    const { body } = await request(app)
      .post('/graphql')
      .send({ query, variables })

    expect(body).to.matchSnapshot()
  })
})
