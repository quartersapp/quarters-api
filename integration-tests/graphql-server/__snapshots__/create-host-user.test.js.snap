// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can create a new host user 1`] = `
Object {
  "data": Object {
    "createHostUser": Object {
      "user": Object {
        "email": "lucas.m@gmail.com",
        "name": "Lucas M.",
      },
    },
  },
}
`;
